Modix/Modules/MoveMessageModule.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Threading.Tasks;
+using Discord;
+using Discord.Commands;
+using Discord.WebSocket;
+using Modix.Services.CommandHelp;
+using Serilog;
+
+namespace Modix.Modules
+{
+    [Name("Move Message"), RequireUserPermission(GuildPermission.ManageMessages), HiddenFromHelp]
+    public class MoveMessageModule : ModuleBase
+    {
+        [Command("move"), Summary("Moves a message from one channel to another."), Remarks("Usage: !move 12345 #foo")]
+        public async Task Run(ulong messageId, SocketTextChannel channel, [Remainder] string reason)
+        {
+            // Ignore bots and same channel-to-channel requests
+            if (Context.User.IsBot || channel.Id == Context.Channel.Id) return;
+
+            IMessage message = null;
+
+            try
+            {
+                message = await GetMessage(messageId, Context.Channel as ITextChannel);
+
+                if (message == null)
+                    message = await FindMessageInUnknownChannel(messageId);
+            }
+            catch (Exception e)
+            {
+                Log.Error(e, "Failed fetching message for Move command, ran by {User} with a Message ID of {MessageId}",
+                    Context.User.Mention, messageId);
+            }
+
+            // Format output
+            var builder = new EmbedBuilder()
+                    .WithColor(new Color(95, 186, 125))
+                    .WithDescription(message.Content);
+
+            if (!string.IsNullOrWhiteSpace(reason))
+            {
+                builder.AddInlineField("Reason", reason);
+            }
+
+            var mover = $"@{(Context.User as SocketGuildUser)?.Nickname ?? Context.User.Username}";
+            builder.WithFooter($"Message copied from #{message.Channel.Name} by {mover}");
+
+            await channel.SendMessageAsync("", embed: builder);
+
+            // Delete the source message, and the command message that started this request
+            await message.DeleteAsync();
+            await Context.Message.DeleteAsync();
+            await Context.Channel.SendMessageAsync($"Message {messageId} moved by {mover}");
+        }
+
+        [Command("move"), Summary("Moves a message from one channel to another."), Remarks("Usage: !move 12345 #foo")]
+        public async Task Run(ulong messageId, SocketTextChannel channel)
+            => await Run(messageId, channel, null);
+
+        [Command("move"), Summary("Moves a message from one channel to another."), Remarks("Usage: !move 12345 #foo")]
+        public async Task Run(SocketTextChannel channel, ulong messageId)
+            => await Run(messageId, channel, null);
+
+        [Command("move"), Summary("Moves a message from one channel to another."), Remarks("Usage: !move 12345 #foo")]
+        public async Task Run(SocketTextChannel channel, ulong messageId, [Remainder] string reason)
+            => await Run(messageId, channel, null);
+
```

Comment:
the duplicate overloads aren't needed.
default reason to null to get rid of the overload without a reason